server.go

```diff
@@ -24,113 +26,140 @@ type Route struct {
 	Params map[string]string `yaml:"-"`
 }
 
+// Routes is a list of Routes.
 type Routes struct {
 	Routes []Route `yaml:"routes"`
 }
 
-type ByPathLength []Route
+type byPathLength []Route
 
-func (b ByPathLength) Len() int           { return len(b) }
-func (b ByPathLength) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
-func (b ByPathLength) Less(i, j int) bool { return len(b[i].Path) > len(b[j].Path) }
+func (b byPathLength) Len() int           { return len(b) }
+func (b byPathLength) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
+func (b byPathLength) Less(i, j int) bool { return len(b[i].Path) > len(b[j].Path) }
 
-const TIMEOUT_MILLISECONDS = 500
+const timeout = 500 * time.Millisecond
 
-var ErrInternalServerGeneric = errors.New("oops, something went wrong")
+var errInternalServerGeneric = errors.New("oops, something went wrong")
 
 func main() {
+	if err := run(); err != nil {
+		log.Fatalf("Error: %v\n", err)
+	}
+}
 
-	cgi_scripts_dir := os.Getenv("CGI_DIR")
-	if cgi_scripts_dir == "" {
-		log.Fatalf("ERROR: CGI_DIR environment variable not set\n")
+func run() error {
+	cgiScriptsDir := os.Getenv("CGI_DIR")
+	if cgiScriptsDir == "" {
+		return fmt.Errorf("CGI_DIR environment variable not set")
 	}
 
-	routes := readAndUnmarshalRoutes(cgi_scripts_dir + "/routes.yaml")
+	routes, err := readAndUnmarshalRoutes(cgiScriptsDir + "/routes.yaml")
+	if err != nil {
+		return fmt.Errorf("read and unmarshal routes: %w", err)
+	}
 
 	// Build each Roc script into an executable
 	for _, route := range routes.Routes {
+		scriptPath := path.Join(cgiScriptsDir, route.Script)
+		binaryPath := path.Join(cgiScriptsDir, route.Binary)
 
-		script_path := path.Join(cgi_scripts_dir, route.Script)
-		binary_path := path.Join(cgi_scripts_dir, route.Binary)
-
-		cmd := exec.Command("roc", "build", script_path)
+		cmd := exec.Command("roc", "build", scriptPath)
 		// --optimize runs slow on Roc PG for now
 		// cmd := exec.Command("roc", "build", "--optimize", script_path)
-		err := cmd.Run()
-		if err != nil {
-			log.Fatalf("ERROR: Unable to build %s: %s", script_path, err.Error())
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("unable to build %s: %w", scriptPath, err)
 		}
 
 		// Check that the executable exists with the expected name
-		if _, err := os.Stat(binary_path); os.IsNotExist(err) {
-			log.Fatalf("ERROR: Expected binary %s does not exist", binary_path)
+		if _, err := os.Stat(binaryPath); errors.Is(err, fs.ErrNotExist) {
+			return fmt.Errorf("expected binary %s does not exist", binaryPath)
 		}
-
 	}
 
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-
 		start := time.Now()
 
-		handleHTTPRequest(w, r, routes, cgi_scripts_dir)
+		if err := handleHTTPRequest(w, r, routes, cgiScriptsDir); err != nil {
```

Comment:
Here is the same. When the handle function returns an error, than it can be handled at one place.